src/index.js

```diff
@@ -1750,14 +1783,57 @@ export class ULabel {
         ctx.imageSmoothingEnabled = false;
         ctx.globalCompositeOperation = "source-over";
 
-        // Draw the box
-        const pts = annotation_object["spatial_payload"];
-        ctx.beginPath();
-        ctx.moveTo((pts[0][0] + diffX) * px_per_px, (pts[0][1] + diffY) * px_per_px);
-        for (var pti = 1; pti < pts.length; pti++) {
-            ctx.lineTo((pts[pti][0] + diffX) * px_per_px, (pts[pti][1] + diffY) * px_per_px);
+        const spatial_type = annotation_object["spatial_type"];
+        let spatial_payload = annotation_object["spatial_payload"];
+        let active_spatial_payload = spatial_payload;
+
+        // if a polygon, n_iters is the length the spatial payload
+        // else n_iters is 1
+        let n_iters = spatial_type === "polygon" ? spatial_payload.length : 1;
+
+        // Draw all polygons/polylines
+        for (let i = 0; i < n_iters; i++) {
+            if (spatial_type === "polygon") {
+                active_spatial_payload = spatial_payload[i];
+            }
+            // Draw the borders
+            const pts = active_spatial_payload;
+            if (pts.length > 0) {
+                ctx.beginPath();
+                ctx.moveTo((pts[0][0] + diffX) * px_per_px, (pts[0][1] + diffY) * px_per_px);
+                for (let pti = 1; pti < pts.length; pti++) {
+                    ctx.lineTo((pts[pti][0] + diffX) * px_per_px, (pts[pti][1] + diffY) * px_per_px);
+                }
+                ctx.stroke();
+            }
+
+            // If polygon is closed, fill it
+            if (spatial_type === "polygon" && GeometricUtils.is_polygon_closed(active_spatial_payload)) {
+                ctx.closePath();
+                ctx.globalAlpha = 0.2;
+                ctx.fill();
+                ctx.globalAlpha = 1.0;
+            }           
+        }
+
+        // For polygons, go back through and unfill the intersectoions of all the polygons
```

Comment:
```suggestion
        // For polygons, go back through and unfill the intersections of all the polygons
```
Is "intersectoions" correct? It was changed to "intersections" in [7b2e4f1](https://github.com/SenteraLLC/ulabel/pull/129/commits/7b2e4f14cb45bbd66e4e16ce77408328c32731ef) and then reverted back to "intersectoions" in [2153662](https://github.com/SenteraLLC/ulabel/pull/129/commits/215366276216ff9c7e8a6ce079090a59c5187359)